gap_analysis/models/business.py

```diff
@@ -0,0 +1,212 @@
+    )
+    odoo_scenario = fields.Text(
+        'Odoo Scenario',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    odoo_gap = fields.Text(
+        'Odoo Gap',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    drop = fields.Boolean(
+        string='Drop?',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="""Determine whether this business requirement is to be kept
+        or dropped."""
+    )
+    rough_estimation_lines = fields.One2many(
+        comodel_name='business.estimation.line',
+        inverse_name='business_requirement_id',
+        string='Rough Estimation Lines',
+        copy=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    categ_id = fields.Many2one(
+        comodel_name='business.requirement.category',
+        string='Category',
+        ondelete='restrict',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    estimated_time_total = fields.Float(
+        compute='_get_estimated_time_total',
+        string='Total Estimated Time',
+        store=True,
+    )
+    state = fields.Selection(
+        selection="_get_states",
+        string='State',
+        default='draft',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+
+    business_requirement_ids = fields.One2many(
+        comodel_name='business.requirement',
+        inverse_name='parent_id',
+        string='Rough Estimation Lines',
+        copy=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    parent_id = fields.Many2one(
+        comodel_name='business.requirement',
+        string='Parent',
+        ondelete='set null',
+        domain="[('id', '!=', id)]"
+    )
+    level = fields.Integer(
+        compute='_get_level',
+        string='Level',
+        store=True
+    )
+    summary_estimation = fields.Boolean(
+        string='Summary Estimation?',
+        default=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    summary_estimation_note = fields.Text(
+        'Summary Estimation Note',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+
+    @api.one
+    @api.depends(
+        'rough_estimation_lines.estimated_time')
+    def _get_estimated_time_total(self):
+        time_total = 0
+        for line in self.rough_estimation_lines:
+            time_total += line.estimated_time
+        self.estimated_time_total = time_total
+
+    @api.multi
+    @api.depends(
+        'parent_id')
+    def _get_level(self):
+        for br in self:
+            level = br._compute_level()
+            br.level = level
+
+    @api.multi
+    def _compute_level(self):
+        level = 1
+        if self.parent_id:
+            level += self.parent_id._compute_level()
+        return level
```

Comment:
SInce `_compute_level` is just a helper for `_get_level` I would prefer for it to be nested insode `_get_level`.
It would make the API cleaner.